lockfile/src/parse_depfile.rs

```diff
@@ -0,0 +1,250 @@
+//! Parse generic dependency files.
+use std::ffi::OsStr;
+#[cfg(feature = "generator")]
+use std::path::Path;
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Context};
+use phylum_types::types::package::{PackageDescriptor, PackageDescriptorAndLockfile};
+use serde::{Deserialize, Serialize};
+
+use crate::{LockfileFormat, Package, PackageVersion, Parse, ThirdPartyVersion};
+
+/// Lockfile parsing error.
+#[derive(thiserror::Error, Debug)]
+pub enum ParseError {
+    /// Dependency file is a manifest, but lockfile generation is disabled.
+    #[error("Parsing {0:?} requires lockfile generation, but it was disabled through the CLI")]
+    ManifestWithoutGeneration(String),
+    #[error(transparent)]
+    Other(#[from] anyhow::Error),
+}
+
+/// Lockfile with all its dependencies.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ParsedLockfile {
+    /// User-facing lockfile identifier (i.e. a file path or name).
+    #[serde(alias = "path")]
+    pub id: String,
```

Comment:
I would rather keep the name `path` for this variable.

Also, since this is only user-facing, we might be able to avoid it entirely in this crate. Currently, it's only used for error messages and `.api_packages()` (which isn't used directly by this crate).

Perhaps we could have functions in this crate return `Vec<PackageDescriptor>` instead of using this struct. Alternately, we could just remove the `id` field from this struct. CLI would then need its own struct to add the `path` where needed.

Reply:
I'd rather have this struct in a central location where it is easy to share between other crates. That way we won't have different decisions for this problem in every single crate.

Reply:
Really not a fan of what this did to our `generate_lockfile` method.

Reply:
I agree. But I think `fs_path` and `display_path` are well-named given the situation.